modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractPythonCodegen.java

```diff
@@ -1994,4 +1434,799 @@ public String toEnumDefaultValue(String value, String datatype) {
+                if (enumVars != null) {
+                    for (Map<String, Object> enumVar : enumVars) {
+                        values.add((String) enumVar.get("value"));
+                    }
+                }
+                return String.format(Locale.ROOT, "%sEnum", cp.nameInCamelCase);
+            } else*/
+
+            if (result == null) {
+                // TODO: Cleanup
+                if (!cp.isPrimitiveType || cp.isModel) { // model
+                    // skip import if it's a circular reference
+                    if (classname == null) {
+                        // for parameter model, import directly
+                        hasModelsToImport = true;
+                        modelImports.add(cp.getDataType());
+                        exampleImports.add(cp.getDataType());
+                    } else {
+                        if (circularImports.containsKey(cp.getDataType())) {
+                            if (circularImports.get(cp.getDataType()).contains(classname)) {
+                                hasModelsToImport = true;
+                                postponedModelImports.add(cp.getDataType());
+                                postponedExampleImports.add(cp.getDataType());
+                                // cp.getDataType() import map of set contains this model (classname), don't import
+                                LOGGER.debug("Skipped importing {} in {} due to circular import.", cp.getDataType(), classname);
+                            } else {
+                                // not circular import, so ok to import it
+                                hasModelsToImport = true;
+                                modelImports.add(cp.getDataType());
+                                exampleImports.add(cp.getDataType());
+                            }
+                        } else {
+                            LOGGER.error("Failed to look up {} from the imports (map of set) of models.", cp.getDataType());
+                        }
+                    }
+                    result = new PythonType(cp.getDataType());
+                } else {
+                    throw new RuntimeException("Error! Codegen Property not yet supported in getPydanticType: " + cp);
+                }
+            }
+
+            return result;
+        }
+
+        private String finalizeType(CodegenProperty cp, PythonType pt) {
+            if (!cp.required || cp.isNullable) {
+                typingImports.add("Optional");
+                PythonType opt = new PythonType("Optional");
+                opt.addTypeParam(pt);
+                pt = opt;
+            }
+
+            if (!StringUtils.isEmpty(cp.description)) { // has description
+                pt.annotate("description", cp.description);
+            }
+
+            // field
+            if (cp.baseName != null && !cp.baseName.equals(cp.name)) { // base name not the same as name
+                pt.annotate("serialization_alias", cp.baseName);
+            }
```

Comment:
This used to be `alias`, but in Pydantic v2, [`alias` is both a validator and a serializer](https://docs.pydantic.dev/2.3/usage/fields/#field-aliases).

From the tests, I assume `alias` was used for serialization only (to produce the aliased field), not for validation (we still want to be able to read the original Pythonic field)

Note that in Pydantic v1, both forms (the field name and the alias name) could be used to create a new instance of the model. This is not possible anymore, only the field name is accepted.